plugins/modules/zos_job_query.py

```diff
@@ -208,53 +208,86 @@ def run_module():
         return result
 
     try:
-        validate_arguments(module.params)
-        jobs_raw = query_jobs(module.params)
-        jobs = parsing_jobs(jobs_raw)
+        name, id, owner = validate_arguments(module.params)
+        jobs_raw = query_jobs(name, id, owner)
+        if jobs_raw:
+            jobs = parsing_jobs(jobs_raw)
+        else:
+            jobs = None
+
     except Exception as e:
         module.fail_json(msg=e, **result)
     result["jobs"] = jobs
     module.exit_json(**result)
 
 
+# validate_arguments rturns a tuple, so we don't have to rebuild the job_name string
 def validate_arguments(params):
     job_name_in = params.get("job_name")
+
     job_id = params.get("job_id")
+
     owner = params.get("owner")
     if job_name_in or job_id:
         if job_name_in and job_name_in != "*":
             job_name_pattern = re.compile(r"^[a-zA-Z$#@%][0-9a-zA-Z$#@%]{0,7}$")
             job_name_pattern_with_star = re.compile(
                 r"^[a-zA-Z$#@%][0-9a-zA-Z$#@%]{0,6}\*$"
             )
-            m = job_name_pattern.search(job_name_in)
-            n = job_name_pattern_with_star.search(job_name_in)
-            if m or n:
-                pass
-            else:
-                raise RuntimeError("Failed to validate the job name: " + job_name_in)
+            test_basic = job_name_pattern.search(job_name_in)
+            test_star = job_name_pattern_with_star.search(job_name_in)
+            # logic twist: o must be non-null value from m or n
+            test_result = test_basic
+            if test_star:
+                test_result = test_star
+
+            job_name_short = "unused"
+            # if neither m nor n were non-null, check if the string needed to be truncated to the first *
```

Comment:
m/n comments don't map to the updated code , its minor just might be helpful for someone in the future.